Refactor of contribute pages (#1437)
elixir-europe/rdmkit: pages/contribute/github_way.md

---
title: GitHub way (beginner)
toc: false
---

This guide tells you how to request and edit a page using GitHub.

Prerequisites:
* No technical knowledge

> * A GitHub account. If you don’t have one, [create a free GitHub account](https://github.com/join) before you start.

Reviewer: ```suggestion
* A GitHub account. If you do not have one, [create a free GitHub account](https://github.com/join) before you start.
```

## Announce and discuss your proposal through GitHub “issues”

1. Go to the [RDMkit repo](https://github.com/elixir-europe/rdmkit) on GitHub.
2. Click “Issues” in the top menu bar and look at the existing issues. See if your idea or suggestion is already being discussed.
    * If an issue exists, comment and let people know if you want to contribute.

>     * If no relevant issue exists, create a new issue by clicking the green “New issue” button on the right and choosing one of the issue templates. You can find more information on creating issues in the [GitHub documentation](https://docs.github.com/en/github/managing-your-work-on-github/creating-an-issue).

Reviewer 2: > ...and choosing one of the issue templates.

...and choose one of the issue templates.

"Choose" needs to be imperative, I think, like "create" before it.

Reviewer: I think the intended coordination is between "clicking" and "choosing", not between "create" and "choosing" (create by clicking and choosing).

Author: @martin-nc do you agree here with @bianchini88 ?

Reviewer 2: Oh, by the time I'd got to "choosing" I'd forgotten about "create"! I suppose you could say:

> If no relevant issue exists, create a new issue: click the green “New issue” button on the right and choose one of the issue templates.

Or just leave it as it is.

Author: Sounds good to me!

3. Discuss your idea with the editors through comments in the issues – you will be notified when others comment.
    * You can read the comments and write your opinion/questions/answers in the “Leave a comment” box. To submit your responses, click the green “Comment” button on the right.
    * You can always return to your opened issue by going to the [issues section](https://github.com/elixir-europe/rdmkit/issues) of our GitHub repo.

{% include callout.html type="tip" content="You can also get to the [RDMkit ](https://github.com/elixir-europe/rdmkit)repo on GitHub using the ‘GitHub’ link in the header of this site." %}

## Read the guides {#read-the-guides}

Before starting editing on GitHub:
1. Make sure you are following our [style guide](https://rdmkit.elixir-europe.org/style_guide).
2. Follow the structure of the provided template for the page you wish to create or update. 

> 3. We use markdown. To know how to create paragraphs, headings, format text, add links and images and much more, follow our [markdown cheat sheet](https://rdmkit.elixir-europe.org/markdown_cheat_sheet).

Reviewer 2: > To know how...

To learn how...

> 3. We use markdown. To know how to create paragraphs, headings, format text, add links and images and much more, follow our [markdown cheat sheet](https://rdmkit.elixir-europe.org/markdown_cheat_sheet).

Reviewer: ```suggestion
3. We use markdown. To learn how to create paragraphs, headings, format text, add links and images and much more, follow our [markdown cheat sheet](https://rdmkit.elixir-europe.org/markdown_cheat_sheet).
```

---

This guide tells you how to request and edit a page using GitHub.

Prerequisites:
* No technical knowledge
* A GitHub account. If you do not have one, [create a free GitHub account](https://github.com/join) before you start.

## Announce and discuss your proposal through GitHub “issues”

1. Go to the [RDMkit repo](https://github.com/elixir-europe/rdmkit) on GitHub.
2. Click “Issues” in the top menu bar and look at the existing issues. See if your idea or suggestion is already being discussed.
    * If an issue exists, comment and let people know if you want to contribute.

>     * If no relevant issue exists, create a new issue by clicking the green “New issue” button on the right and choosing one of the issue templates. You can find more information on creating issues in the [GitHub documentation](https://docs.github.com/en/github/managing-your-work-on-github/creating-an-issue).

Reviewer: ```suggestion
    * If no relevant issue exists, create a new one: click the green “New issue” button on the right and choose a template. You can find more information on creating issues in the [GitHub documentation](https://docs.github.com/en/github/managing-your-work-on-github/creating-an-issue).
```

Reviewer: I implemented @martin-nc's suggestion above, as it is clearer. I made additional modifications to avoid repeating the word "issue" too often.

Author: Woops didn't see this comment, I've also implemented it !
